### PR TITLE
1705/m/web 2443.web 2960 truncate columns and persist widths

### PR DESCRIPTION
**Pull request by alecjacobs5401**
@nburwell 

https://ringrevenue.atlassian.net/browse/WEB-2960

Mind giving this a review? Might be beneficial to get this reviewed so I can bump the version on this branch for QA

Includes the following:

1.  `bTruncateHeaders` flag that is defaulted to true
2. Truncate headers when `bTruncateHeaders` flag is enabled
3. Scroll table when resizing columns

### CODE REVIEW

File: src/plugins/data_table/dataTables.colReorder.js [hunk 1]
```diff
@@ -519,8 +527,10 @@
     ColReorder.aoInstances.push(this);
 
 
+    var tableElement = $(this.s.dt.nTable);
+
     if (this.s.bResizeTableWrapper) {
```
Comment (nburwell) [inline src/plugins/data_table/dataTables.colReorder.js]: this boolean is called resizeTableWrapper but now we're resizing the element itself .. is that desired? Should we rename this option?

File: src/plugins/data_table/dataTables.colReorder.js [hunk 2]
```diff
@@ -519,8 +527,10 @@
...
-      $(this.s.dt.nTableWrapper).width($(this.s.dt.nTable).width());
+      tableElement.width(tableElement.width());
```
Comment (nburwell) [inline src/plugins/data_table/dataTables.colReorder.js]: and setting it's width... to it's width?  what does this do, and if it's on purpose can you add a comment please?

File: src/plugins/data_table/dataTables.colReorder.js [hunk 3]
```diff
@@ -534,9 +544,18 @@
     // fix the width and add table layout fixed.
     if (this.s.bAddFixed) {
       // Set the table to minimum size so that it doesn't stretch too far
-      $(this.s.dt.nTable).width("10px");
-      $(this.s.dt.nTable).width($(this.s.dt.nTable).width()).css('table-layout','fixed');
+      tableElement.width("10px");
+      tableElement.width(tableElement.width()).css('table-layout','fixed');
```
Comment (nburwell) [inline src/plugins/data_table/dataTables.colReorder.js]: no wonder you were gouging your eyes out...

File: src/plugins/data_table/dataTables.colReorder.js [hunk 4]
```diff
@@ -735,6 +754,10 @@
         this.s.bResizeTableWrapper = this.s.init.bResizeTableWrapper;
       }
 
+      if (typeof this.s.init.bTruncateHeaders != 'undefined') {
```
Comment (nburwell) [inline src/plugins/data_table/dataTables.colReorder.js]: `!==` please

Comment (nburwell) [inline src/plugins/data_table/dataTables.colReorder.js]: and maybe the surrounding options too..

File: src/plugins/data_table/data_table.js [hunk 5]
```diff
@@ -347,6 +347,11 @@ var LocalDataTable = (function() {
       }
     },
 
+    _columnElements: function(selector) {
+      var selectorString = selector || "";
+      return this.$("table").find("thead tr th" + selectorString);
+    },
```
Comment (nburwell) [inline src/plugins/data_table/data_table.js]: is this method used?

File: src/plugins/data_table/dataTables.colReorder.js (hunk 2 above)
Comment (alecjacobs5401) [inline src/plugins/data_table/dataTables.colReorder.js]: Oh thanks for catching this...I _think_ that I was trying to DRY out all the `$(this.s.dt.nTable)`s and replaced one too many

File: src/plugins/data_table/dataTables.colReorder.js (hunk 4 above)
Comment (alecjacobs5401) [inline src/plugins/data_table/dataTables.colReorder.js]: Ill go through and cleanup all of declarations

File: src/plugins/data_table/data_table.js (hunk 5 above)
Comment (alecjacobs5401) [inline src/plugins/data_table/data_table.js]: Yeah, I was using it in action table, but I felt that the method was better suited here instead. I'll make sure I had added tests for this

Comment (nburwell) [inline src/plugins/data_table/data_table.js]: ah got it.  Well if it's used there can we remove the underscore please?  (I know we haven't been strict on that everywhere in backdraft but it's definitely preferred to indicate it's a "public" API)

Comment (alecjacobs5401) [inline src/plugins/data_table/data_table.js]: Definitely, I'll be sure to be more aware of that in the future as well

**Comment by alecjacobs5401** [comment]
@nburwell 

Comments addressed, ready for another round

**Comment by nburwell** [comment]
👍

**Pull request merged**